mne/beamformer/_lcmv.py

```diff
@@ -50,6 +50,19 @@ def _reg_pinv(x, reg):
     return linalg.pinv(x), d
 
 
+def _eig_inv(x, rank):
+    """Compute a pseudoinverse with smallest component set to zero."""
+    U, s, V = linalg.svd(x)
+
+    # pseudoinverse is computed by setting eigenvalues not included in
+    # signalspace to zero
+    s_inv = np.zeros(s.shape)
+    s_inv[:rank] = 1 / s[:rank]
+
+    x_inv = np.dot(V.T, np.dot(np.diag(s_inv), U.T))
```

Comment:
you do not need to do `np.diag`, `np.dot(np.diag(s_inv), U.T)` is equivalent to `s_inv[:, np.newaxis] * U.T` (and you can even make `s_inv = np.zeros((s.size, 1)) above` then you don't need the `np.newaxis`)